system_tests/espresso_osp_test.go

```diff
@@ -0,0 +1,98 @@
+package arbtest
+
+import (
+	"context"
+	"encoding/json"
+	"math/big"
+	"os"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/offchainlabs/nitro/solgen/go/ospgen"
+	"github.com/offchainlabs/nitro/solgen/go/test_helpersgen"
+	"github.com/offchainlabs/nitro/validator"
+	"github.com/offchainlabs/nitro/validator/server_arb"
+	"github.com/offchainlabs/nitro/validator/server_common"
+)
+
+func TestEspressoOsp(t *testing.T) {
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	initialBalance := new(big.Int).Lsh(big.NewInt(1), 200)
+	l1Info := NewL1TestInfo(t)
+	l1Info.GenerateGenesisAccount("deployer", initialBalance)
+
+	deployerTxOpts := l1Info.GetDefaultTransactOpts("deployer", ctx)
+
+	chainConfig := params.ArbitrumDevTestChainConfig()
+	l1Info, l1Backend, _, _ := createTestL1BlockChain(t, l1Info)
+	hotshotAddr, tx, hotShotConn, err := test_helpersgen.DeployMockHotShot(&deployerTxOpts, l1Backend)
+	Require(t, err)
+	_, err = EnsureTxSucceeded(ctx, l1Backend, tx)
+	Require(t, err)
+
+	rollup, _ := DeployOnTestL1(t, ctx, l1Info, l1Backend, chainConfig, hotshotAddr)
+
+	ospEntryAddr := common.HexToAddress("0xffd0c2C95214aa9980D7419bd87c260C80Ce2546")
+
+	locator, err := server_common.NewMachineLocator("")
+	if err != nil {
+		Fatal(t, err)
+	}
+	wasmModuleRoot := locator.LatestWasmModuleRoot()
+	if (wasmModuleRoot == common.Hash{}) {
+		Fatal(t, "latest machine not found")
+	}
+
+	data, err := os.ReadFile("espresso-e2e/validation_input.json")
```

Comment:
Could you add a comment on how the data in `validation_input.json` is obtained so we don't have to start from scratch if we need to update it?